packages/calcite-components/src/components/chip/chip.scss

```diff
@@ -1,357 +1,405 @@
-.container,
-.select-icon,
-.title {
-  padding-inline-start: var(--calcite-internal-chip-padding-start);
-  padding-inline-end: var(--calcite-internal-chip-padding-end);
-}
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-chip-background-color: Specifies the background color of the component.
+ * @prop --calcite-chip-border-color: Specifies the border color of the component.
+ * @prop --calcite-chip-corner-radius: Specifies the border radius of the component.
+ * @prop --calcite-chip-text-color: Specifies the text color of the component.
+ *
```

Comment:
I think we'd at the very least want to add `--calcite-chip-icon-color` and `--calcite-chip-selection-icon-color` (or whatever name) from the 7180 PR: https://github.com/Esri/calcite-design-system/blob/2c5e985b86621e044f60c94269edab7f83573725/packages/calcite-components/src/components/chip/chip.scss#L6

We can't just set `--calcite-icon-color` on a Calcite Chip - it has too many internally rendered icons that would be affected. The 7180 PR also had `--calcite-chip-close-icon-color` (and, props to control internal close button states, which it sounds like we did not want to pursue - I still think this prevents a lot of valid use cases)

Reply:
The design pattern for calcite-chip is for icons to follow the text color. If a user wants to have a special color they can do something like `calcite-chip.my-chip { --calicte-icon-color: red }`.  Similarly. IMO, all close sub-elements (should) have the same interaction pattern with a shared token set. Something like

```
--calcite-close-color
--calcite-close-background-color-hover
--calcite-close-background-color-press
```

Reply:
Agreed they should be (ideally) consistently implemented when overridden - but not adding component-specific ones (even if it falls back to the global one) to the component means they won't be documented alongside all the other component tokens - wherever that doc is consumed (including places we can't manually supplement it with info about these generic "close-color" props).

Currently it seems like `text-color` property here also sets the selection affordance icon as well as the `icon-start` icon. At the very least I'd like to be able to not have this happen (which is the current behavior): 

The current behavior that is available (just overriding the global css prop - changes the affordance as well as the icon, which I don't think is always preferable). 

![Screenshot 2024-08-28 at 1 57 43 PM](https://github.com/user-attachments/assets/dc49aa7d-9bee-4248-a3f8-c3e4d90bd536)

My use case for overriding an icon color is probably to match map symbology or layer display color / graphics. So I don't think I'd want the selection icon to change as well:

![Screenshot 2024-08-28 at 1 57 54 PM](https://github.com/user-attachments/assets/b60af8ef-16d0-4938-8261-20334783b5ed)